### PR TITLE
SONARIAC-443 Docker: parse FROM instruction

### PR DESCRIPTION
**Pull request by nils-werner-sonarsource**

### CODE REVIEW

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/ParamTreeImpl.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import org.sonar.iac.docker.tree.api.ParamTree;
+import org.sonar.iac.docker.tree.api.SyntaxToken;
+
+public class ParamTreeImpl extends DockerTreeImpl implements ParamTree {
+
+  private final SyntaxToken prefix;
+  private final SyntaxToken name;
+  private final SyntaxToken equals;
+  private final SyntaxToken value;
+
```
Comment (rudy-regazzoni-sonarsource) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/ParamTreeImpl.java:33]: Look very much like the new version of KeyValuePair in PR ADD, it has the almost same parameters prefix / key / equals / value (most of them optional).

I guess we could either switch to it or keep KeyValuePair for cases without prefix and use ParamTree in ADD.

**Review by rudy-regazzoni-sonarsource — APPROVED** [review]
LGTM ! Please see my comments.

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerGrammar.java [hunk 2]
```diff
@@ -81,19 +87,30 @@ public FromTree FROM() {
...
-      f.keyValuePairEquals(
-        b.token(DockerLexicalGrammar.PLATFORM),
-        b.token(Punctuator.EQU),
-        b.token(DockerLexicalGrammar.STRING_LITERAL_WITHOUT_SPACE)
+  public ParamTree PARAM() {
+    return b.<ParamTree>nonterminal(DockerLexicalGrammar.PARAM).is(
+      f.param(
+        b.token(DockerLexicalGrammar.PARAM_PREFIX),
+        b.token(DockerLexicalGrammar.PARAM_NAME),
+        b.token(DockerLexicalGrammar.EQUALS_OPERATOR),
```
[the remark this answers is not in the document]
Comment (nils-werner-sonarsource) [reply iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerGrammar.java:102]: For now, all elements from `Punctuator` are prefixed with a space. There should not be a space before the equals sign for an instruction parameter.

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/ParamTreeImpl.java (hunk 1 above)
Comment (nils-werner-sonarsource) [reply iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/ParamTreeImpl.java:33]: I would like to replace all `KeyValuePair` usage with a more precise `ParamTree` where applicable. However, there are cases, where we will need this `KeyValuePair`.

**Pull request merged**